components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -1063,6 +1063,10 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
                 }).catch();
             }
 
+            if (User.isOnboardingUser(user) && !IDESpecifiedContext.is(context)) {
+                return { showUserIdePreference: true };
```

Comment:
Why sending back such a dashboard-specific signal?
That doesn't fit into the existing `WorkspaceCreationResult` shape. 

Is it possible to rewrite in a way, that this is checked on frontend, so that `createWorkspace` doesn't need to be changed.

Reply:
see `ContextURL` from https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/src/context-url.ts

Reply:
It can be here, but the check will be more weak because prefix maybe failed in server regex

Reply:
Which regex? 

The question is, can we test `User.isOnboardingUser(user) && !IDESpecifiedContext.is(context)` on frontend? 
If so, let's do it, and avoid an extra roundtrip to server.

Reply:
regex like this https://github.com/gitpod-io/gitpod/blob/45f13cf8e9214db664adc23a418e53d92364fbb7/components/server/src/workspace/referrer-prefix-context-parser.ts#L13-L17

I will go with check URL prefix on frontend only